scripts/lib/CIME/SystemTests/system_tests_common.py

```diff
@@ -478,14 +478,14 @@ def _compare_baseline(self):
         """
         with self._test_status:
             # compare baseline
-            success, comments = compare_baseline(self._case)
+            success, comments, ts_comments = compare_baseline(self._case)
             append_testlog(comments)
             status = TEST_PASS_STATUS if success else TEST_FAIL_STATUS
             baseline_name = self._case.get_value("BASECMP_CASE")
-            ts_comments = (os.path.dirname(baseline_name) + ": " + comments) if "\n" not in comments else os.path.dirname(baseline_name)
-            if ts_comments == os.path.dirname(baseline_name) and TEST_NO_BASELINES_COMMENT in comments:
-                ts_comments += " {} some baseline files were missing".format(TEST_NO_BASELINES_COMMENT)
-
+            if ts_comments:
```

Comment:
I don't think that's true: There are four callers of compare_baseline:

1. This system_test_common call, which cares about ts_comments

2. `scripts/lib/CIME/compare_test_results.py`, which also cares about ts_comments. (This was one of my motivations for making this a return of `compare_baseline`, because otherwise it seemed there would need to be some duplicated logic in this and (1).)

3. `scripts/lib/CIME/bless_test_results.py`, which doesn't seem to need `ts_comments`, though I honestly haven't looked closely.

4. `scripts/Tools/component_compare_baseline`... is this even used any more?


That said, I'm fine with your suggestion of `get_ts_synopsis(comments)` if you can see a good way to implement it. I'd be concerned about its fragility, since the full comments can contain pretty much anything, but if you can come up with a way to do it (maybe putting the `ts_synopsis` part within some delimiters???), then I'm fine with it.

Reply:
I use component_compare_baseline from the command line occasionally.